shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/domain/analytic/StepCompletionShareStreakModalShownHyperskillAnalyticEvent.kt

```diff
@@ -0,0 +1,41 @@
+package org.hyperskill.app.step_completion.domain.analytic
+
+import org.hyperskill.app.analytic.domain.model.hyperskill.HyperskillAnalyticAction
+import org.hyperskill.app.analytic.domain.model.hyperskill.HyperskillAnalyticEvent
+import org.hyperskill.app.analytic.domain.model.hyperskill.HyperskillAnalyticPart
+import org.hyperskill.app.analytic.domain.model.hyperskill.HyperskillAnalyticRoute
+import org.hyperskill.app.analytic.domain.model.hyperskill.HyperskillAnalyticTarget
+
+/**
+ * Represents a shown analytic event of the share streak modal.
+ *
+ * JSON payload:
+ * ```
+ * {
+ *     "route": "/learn/step/1",
+ *     "action": "shown",
+ *     "part": "modal",
+ *     "target": "share_streak_modal",
+ *     "context":
+ *     {
+ *         "streak": 1
+ *     }
+ * }
+ * ```
+ *
+ * @see HyperskillAnalyticEvent
+ */
+class StepCompletionShareStreakModalShownHyperskillAnalyticEvent(
+    route: HyperskillAnalyticRoute,
+    val streak: Int
+) : HyperskillAnalyticEvent(
+    route,
+    HyperskillAnalyticAction.SHOWN,
+    HyperskillAnalyticPart.MODAL,
+    HyperskillAnalyticTarget.SHARE_STREAK_MODAL
+) {
+    override val params: Map<String, Any>
+        get() = super.params + mapOf(
+            PARAM_CONTEXT to mapOf(StepCompletionHyperskillAnalyticParams.PARAM_STREAK to streak)
+        )
```

Comment:
It's a really nice idea to pass streak into params - maybe we will see some correlation between streak number and conversion to sharing 👍